direct/tpm2/tpm2.go

```diff
@@ -290,6 +290,38 @@ type LoadExternalResponse struct {
 // Response implements the Response interface.
 func (*LoadExternalResponse) Response() tpm.CC { return tpm.CCLoadExternal }
 
+// ReadPublic is the input to TPM2_ReadPublic.
+// See definition in Part 3, Commands, section 12.4
+type ReadPublic struct {
+	// tpm handle of an object
+	ObjectHandle tpmi.DHObject `gotpm:"handle"`
+}
+
+// Command implements the Command interface.
+func (*ReadPublic) Command() tpm.CC { return tpm.CCReadPublic }
+
+// Execute executes the command and returns the response.
+func (cmd *ReadPublic) Execute(t transport.TPM, s ...Session) (*ReadPublicResponse, error) {
+	var rsp ReadPublicResponse
+	if err := execute(t, cmd, &rsp, s...); err != nil {
+		return nil, err
+	}
+	return &rsp, nil
+}
+
+// ReadPublicResponse is the response from TPM2_ReadPublic.
+type ReadPublicResponse struct {
+	// public area of an object
```

Comment:
nit: can these comments be verbatim copies of the "description" column in the spec?